src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from 'react';
+import {
+  SelectableBox,
+  Icon,
+  Spinner,
+} from '@edx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { ArrowDropDown, ArrowDropUp } from '@edx/paragon/icons';
+import PropTypes from 'prop-types';
+import messages from './messages';
+import './ContentTagsDropDownSelector.scss';
+
+import { useTaxonomyTagsDataResponse, useIsTaxonomyTagsDataLoaded } from './api/hooks/selectors';
+
+const ContentTagsDropDownSelector = ({
+  taxonomyId, level, subTagsUrl,
+}) => {
+  const intl = useIntl();
+  // This array represents the states of the dropdowns on this level
+  // Each index contains whether the dropdown is open/closed
+  // O === Closed
+  // 1 === Open
+  const [dropdownStates, setDropdownStates] = useState([]);
+
+  const isOpen = (i) => (dropdownStates[i] === 1);
+
+  const clickAndEnterHandler = (index) => {
+    const updatedState = dropdownStates.map((dropdownState, i) => {
+      if (index === i) {
+        return dropdownState === 0 ? 1 : 0;
+      }
+      return dropdownState;
+    });
+
+    setDropdownStates(updatedState);
```

Comment:
If I understand correctly, this is just flipping the value at index i?
Why not simplify it as follows:
```suggestion
    const updatedState = [...dropdownStates] // make a copy
    updatedState[i] ^=1; // flip 1 to 0 or 0 to 1 or use ternirary or if swithing to true false just use = !...
    setDropdownStates(updatedState);
```